devdoc/userdoc/Migrating-from-StdCrypto-to-StdOpenSSL.md

```diff
@@ -0,0 +1,223 @@
+# Migrating from Std.Crypto to Std.OpenSSL
+
+This guide explains how to transition from the deprecated `Std.Crypto` module functions to their replacements in `Std.OpenSSL.*`.
+
+## Overview of the Migration Process
+
+The `Std.Crypto` module provided cryptographic functionality across four main areas:
+
+- Algorithm enumeration (Supported* functions)
+- Hashing operations
+- Symmetric encryption
+- Public key encryption (multiple variants differing by key source and passphrase type)
+
+All cryptographic functionality has been consolidated into clearer, more cohesive namespaces within `Std.OpenSSL.*`. This migration improves API consistency, reduces complexity, and provides better integration with modern OpenSSL features.
+
+## Namespace Migration Mapping
+
+The following table shows how functional areas map from old to new namespaces:
+
+| Functional Area | Old Namespace(s) | New Namespace |
+|-----------------|------------------|---------------|
+| Hash / Digest   | `Std.Crypto.Hashing` | `Std.OpenSSL.Digest` |
+| Symmetric Ciphers | `Std.Crypto.SymmetricEncryption`, `Std.Crypto.SymmEncryption` | `Std.OpenSSL.Ciphers` |
+| Public Key (RSA Seal/Unseal, Sign/Verify) | Multiple: `PublicKeyEncryption*`, `PKEncryption*` (file, LFN, buffer variants) | `Std.OpenSSL.PublicKey` |
+| Algorithm Enumeration | `Std.Crypto.SupportedHashAlgorithms()`, etc. | `Std.OpenSSL.Digest.AvailableAlgorithms()`, `Std.OpenSSL.Ciphers.AvailableAlgorithms()` |
+| Public Key Algorithm Enumeration | `Std.Crypto.SupportedPublicKeyAlgorithms()` (deprecated) | (No direct replacement yet) |
+
+## Function Deprecation Reference
+
+The following table provides a direct mapping from deprecated functions to their replacements:
+
+| Deprecated (Std.Crypto) | Replacement (Std.OpenSSL) |
+|-------------------------|---------------------------|
+| SupportedHashAlgorithms() | Digest.AvailableAlgorithms() |
+| SupportedSymmetricCipherAlgorithms() | Ciphers.AvailableAlgorithms() |
+| SupportedPublicKeyAlgorithms() | (No direct substitute) |
+| Hashing.Hash() | Digest.Hash() |
```

Comment:
I missed this earlier: We should normalize the function names in this section, as to whether or not they have have '()' after them. I think it looks better and is a clearer indication that is *is* a function, but either way will work.